tests/components/todoist/test_calendar.py

```diff
@@ -42,3 +49,36 @@ def test_parse_due_date_without_timezone_uses_offset():
     }
     actual = _parse_due_date(data, timezone_offset=-8)
     assert datetime(2022, 2, 2, 22, 0, 0, tzinfo=dt.UTC) == actual
+
+
+@pytest.fixture(name="state")
+def mock_state() -> dict[str, Any]:
+    """Mock the api state."""
+    return {
+        "collaborators": [],
+        "labels": [{"name": "label1", "id": 1}],
+        "projects": [{"id": 12345, "name": "Name"}],
+    }
+
+
+@patch("homeassistant.components.todoist.calendar.TodoistAPI")
+async def test_calendar_entity_unique_id(todoist_api, hass, state):
```

Comment:
I attempted to write a test for the custom project case to test that it's unique_id is `None` but that entity never shows up in the entity registry.  I can look up its state via `hass.states`, but it wasn't in the registry.  I might be missing awaiting something, but I couldn't figure that out.

Reply:
If the entity doesn't have a unique id it won't be part of the entity registry. Assert that that's true.

Reply:
Oh, that makes sense, thank you!